python/tools/polarfire_esb.py

```
@@ -131,19 +92,28 @@ def _spi_program(hololink):
+            "content"
+        ]  # should be the zip or spi file name
+        current_file_cfg = default_config["hololink"]["content"][fname]
+        filename = current_file_cfg.get("filename", None)
+        current_url = current_file_cfg.get("url", None)
+        if current_url:
+            zipname = current_url.split("/")[-1]
+            lfname = zipname
+        else:
+            lfname = None
+        lfilename = filename
+    except Exception:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)